### PR TITLE
Upgrade Ruma

### PR DESCRIPTION
**Pull request by jplatte**
Since we agreed to get Ruma 0.6.0 out before the next SDK release, we can bump versions more frequently again. This PR had me discover a bunch of small issues in Ruma, which have all been fixed along the way.

Depends on

* https://github.com/ruma/ruma/pull/1021
* https://github.com/ruma/ruma/issues/1024

### CODE REVIEW

**Comment by jplatte** [comment]
This should improve SDK build times because we no longer need to activate unstable-pre-spec which was pulling in a lot of unnecessary stuff due to weak dependency features not being available yet.

File: crates/matrix-sdk-base/src/media.rs
Comment (poljar) [inline crates/matrix-sdk-base/src/media.rs:80]: Can we call this and the accompanying methods `source` instead, I prefer plain English, unless things become too long.

File: crates/matrix-sdk-base/src/media.rs
Comment (poljar) [inline crates/matrix-sdk-base/src/media.rs:222]: I assume some variant of those tests are now part of Ruma?

**Review by poljar — APPROVED** [review]
Looks good, please only merge this after the mentioned Ruma PR has been merged. Just in case, so we don't end up depending on a `rev` that doesn't exits due to rebases.

File: crates/matrix-sdk-base/src/media.rs
Comment (jplatte) [reply crates/matrix-sdk-base/src/media.rs:80]: Sure, though it would be nice to do the same in Ruma then, for consistency. cc @zecakeh wdyt?

File: crates/matrix-sdk-base/src/media.rs
Comment (jplatte) [reply crates/matrix-sdk-base/src/media.rs:222]: I'll make sure we have them.

File: crates/matrix-sdk-base/src/media.rs
Comment (zecakeh) [reply crates/matrix-sdk-base/src/media.rs:80]: Sure, I don't have a strong opinion about this

**Comment by jplatte** [comment]
Concerns resolved, will squash and merge after fixing conflicts.

**Pull request merged**
